tests/test_algorithm.py

```diff
@@ -726,9 +726,37 @@ def test_future_symbol(self):
         with self.assertRaises(TypeError):
             algo.future_symbol({'foo': 'bar'})
 
+    def test_future_chain_offset(self):
+        algo = TradingAlgorithm(env=self.env)
+        algo.datetime = pd.Timestamp('2006-12-01', tz='UTC')
+
+        self.assertEqual(
+            algo.future_chain('CL', offset=1).as_of_date,
+            self.trading_calendar.next_session_label(
+                algo.datetime
+            )
+        )
+
+        self.assertEqual(
+            algo.future_chain("CL", offset=5).as_of_date,
+            pd.Timestamp("2006-12-08", tz='UTC')
```

Comment:
It's a little confusing that `offset=1` is implemented in terms of the calendar, but `offset=5` is hard-coded.

Thoughts on either doing `self.trading_calendar.sessions_window(algo.datetime, 5)[-1]`, or iterating calls to `{next,previous}_session_label`.

e.g.
```
In [20]: def iterate(func, n, arg):
    while n > 0:
        arg = func(arg)
        n -= 1
    return arg
   ....:

In [21]: iterate(cal.previous_session_label, 1, dt)
Out[21]: Timestamp('2016-08-29 00:00:00+0000', tz='UTC', offset='C')

In [22]: iterate(cal.previous_session_label, 5, dt)
Out[22]: Timestamp('2016-08-23 00:00:00+0000', tz='UTC', offset='C')
```